### PR TITLE
Add more accessibility sprinkles

### LINKED ISSUE #1154: Content audit for Accessibility guidance
Let's review our docs content for 

- areas where it makes sense to mention (especially Astro-specific) tips for designing with a11y in mind.
- our code examples to make sure our own suggestions suggest good practices
- ... something else?

Building off of @Yan-Thomas's comments in Discord:

"We could sprinkle some accessibility tips and tricks in some pages, I think we can easily explain some of Astro's features using accessibility. For example, we can showcase how we can easily change an page lang atrribute using layouts and briefly explain why this is helpful. Also worth mentioning that UI framework-specific accessibility patterns should be the same (if the component is properly hydrated)."

### PR DESCRIPTION
**Pull request by Yan-Thomas**
#### What kind of changes does this PR include?

- New or updated content

#### Description

- Closes #1154 

This PR adds more accessibility info to different pages, solving the issues we found in our content accessibility audit.

### CODE REVIEW

File: src/pages/en/core-concepts/framework-components.md [hunk 1]
```diff
@@ -87,6 +87,10 @@ the user scrolls down and the component is visible on the page -->
 Any renderer JS necessary for the component's framework (e.g. React, Svelte) is downloaded with the page. The `client:*` directives only dictate when the _component JS_ is imported and when the _component_ is hydrated.
 :::
 
+:::note[Accessibility]
+Framework-specific accessibility patterns should work the same in Astro. Though, you need to be careful with the client directive being used so your accessibility-related JavaScript is being properly loaded and executed when needed.
```
Comment (sarah11918) [inline src/pages/en/core-concepts/framework-components.md:91]:
```suggestion
Most framework-specific accessibility patterns should work the same when these components are used in Astro. Be sure to choose a client directive that will ensure any accessibility-related JavaScript is properly loaded and executed at the appropriate time!
```

**Review by sarah11918 — COMMENTED** [review]
Great sprinkles, Yan! Made a suggestion!  💜

**Comment by sarah11918** [comment]
LGTM!  💜

**Comment by Yan-Thomas** [comment]
Thanks for the review @sarah11918!

**Pull request merged**